docs/website/docs/general-usage/credentials/config_specs.md

```diff
@@ -0,0 +1,327 @@
+# Parses the native value and updates the credentials.
+native_value = ...
+credentials.parse_native_representation(native_value)
+
+# Retrieve native credentials for Google clients
+# For example, build the service object for Google Analytics PI.
+client = BetaAnalyticsDataClient(credentials=credentials.to_native_credentials())
+
+# Get a string representation of the credentials
+# Returns a string representation of the credentials in the format client_email@project_id.
+credentials_str = str(credentials)
+```
+
+#### GcpOAuthCredentials
+
+The GcpOAuthCredentials class is responsible for handling OAuth2 credentials for
+desktop applications in Google Cloud Platform (GCP).
+It can parse native values either as GoogleOAuth2Credentials or as
+serialized OAuth client secrets JSON.
+This class provides methods for authentication and obtaining access tokens.
+
+Usage:
+```python
+oauth_credentials = GcpOAuthCredentials()
+
+# Accepts a native value, which can be either an instance of GoogleOAuth2Credentials
+# or serialized OAuth client secrets JSON.
+# Parses the native value and updates the credentials.
+native_value_oauth = ...
+oauth_credentials.parse_native_representation(native_value_oauth)
+
+# Authenticate and get access token
+oauth_credentials.auth(scopes=["scope1", "scope2"])
+
+# Retrieve native credentials for Google clients
+# For example, build the service object for Google Analytics PI.
+client = BetaAnalyticsDataClient(credentials=oauth_credentials.to_native_credentials())
+
+# Get a string representation of the credentials
+# Returns a string representation of the credentials in the format client_id@project_id.
+credentials_str = str(oauth_credentials)
+```
+
+[The example](https://github.com/dlt-hub/verified-sources/blob/master/sources/google_analytics/setup_script_gcp_oauth.py): how you can get the refresh token using `dlt.secrets.value`.
+
+### AwsCredentials
+
+The AwsCredentials class is responsible for handling AWS credentials,
+including access keys, session tokens, profile names, region names, and endpoint URLs.
+It inherits the ability to manage default credentials and extends it with methods
+for handling partial credentials and converting credentials to a botocore session.
+
+Usage:
+```python
+credentials = AwsCredentials()
+
+# Set the necessary attributes
+credentials.aws_access_key_id = "ACCESS_KEY_ID"
+credentials.aws_secret_access_key = "SECRET_ACCESS_KEY"
+credentials.region_name = "us-east-1"
+
+# Convert credentials to s3fs format
+s3fs_credentials = credentials.to_s3fs_credentials()
+print(s3fs_credentials["key"])
+
+# Imports an external boto3 session and sets the credentials properties accordingly.
+import botocore.session
+session = botocore.session.get_session()
+credentials.parse_native_representation(session)
+print(credentials.aws_access_key_id)
+
+# Get AWS credentials from botocore session
+aws_credentials = credentials.to_native_credentials()
+print(aws_credentials.access_key)
+```
```

Comment:
1. provide a dlt.source example
2. look at filesystem how the s3fs credentials are obtained
3. native credentials: are credentials as used by botocore session and can be passed directly to it.
4. if configuration is not provided, `dlt` uses the default AWS credentials as present on the machine
- it works by creating an instance of botocore Session
- if `profileName` is specified, the credentials for that profile are used. if not - the default profile is used